### ISSUE TITLE
Login Credentials missing when using azcopy with OAuth

### ISSUE THREAD

**Issue by fabienarnaud** [Azure/azure-storage-azcopy#1393]
### Which version of the AzCopy was used? 
10.9.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run?
C:\Users\sa_SQL_RIBT-DW_2BI\AZcopy\azcopy copy "C:\Users\sa_SQL_RIBT-DW_2BI\BI_exports\IDW_OTHER_DATA_2021-04-08.csv" "https://bdwuploaddev.blob.core.windows.net/baseline"

### What problem was encountered?
When I run this command manually from the Powershell window of the on-premises SQL Server, it works perfectly.
Note that I first logged in using the azcopy login command which generated an accessToken.json file in the .azcopy folder.

However when I try to run the same command from a SQL Server Agent job, I receive the following error:
"Login Credentials missing. No SAS token or OAuth token is present and the resource is not public"
This occurs although the SQL Server Agent job is set to run as the same service account which I use when testing the command from the Powershell window.

### How can we reproduce the problem in the simplest way?
- Run the azcopy login command to generate an OAuth token
- Create a Powershell-type job in SSMS SQL Server Agent that runs an azcopy copy command using OAuth (only indicate the source file and the destination storage account and container)
- Set the SQL Server Agent job to run as the account used to run the azcopy login command
- Run the SQL Server Agent job

### Have you found a mitigation/solution?
The only mitigation found so far is to use a SAS token but I would like to avoid exposing a SAS token in clear text in my script

**Comment by nakulkar-msft** [comment]
@fabienarnaud The 'user' who created the login command might not be the same 'user' who runs the job. Have you tried [auto-login](https://github.com/Azure/azure-storage-azcopy/wiki/Improved-login-support-for-AzCopy-commands-(with-in-memory-secret-store)) ? 'azcopy login' is recommended for interactive workflow and SAS/auto-login for automation.

**Comment by fabienarnaud** [comment]
Hi Nakulkar,

Thanks for your suggestion. Both should have been the same user since I created the login command with the same user defined in the SQL Server Agent job "run as" but somehow in the background I guess that indeed this was not the case.
I rewrote my script outside of SSMS SQL Server Agent, directly in a ps1 file, and scheduled it using the Windows Task Scheduler and now the azcopy works perfectly fine!
